v2/internal/attacktechniques/aws/persistence/sts-federation-token/main.go

```diff
@@ -0,0 +1,119 @@
+package aws
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus/mitreattack"
+	"log"
+
+)
+
+//go:embed main.tf
+var tf []byte
+
+func init() {
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:           "aws.persistence.sts-federation-token",
+		FriendlyName: "Generation of AWS temporary keys from IAM credentials",
+		Description: `
+Establishes persistence by generating new AWS temporary keys that remain functional even if the original IAM user is blocked.
+
+Warm-up: 
+
+- Create an IAM user and generate a pair of access keys.
+
+Detonation: 
+
+- Use the access keys from the IAM user to request temporary security credentials via AWS STS.
+- Call the sts:GetCallerIdentity API to validate the usage of the new temporary credentials and ensure they are functional.
+
+References:
+
+- https://www.crowdstrike.com/en-us/blog/how-adversaries-persist-with-aws-user-federation/
+- https://reinforce.awsevents.com/content/dam/reinforce/2024/slides/TDR432_New-tactics-and-techniques-for-proactive-threat-detection.pdf
+- https://fwdcloudsec.org/assets/presentations/2024/europe/sebastian-walla-cloud-conscious-tactics-techniques-and-procedures-an-overview.pdf
+`,
+		Detection: `
+Through CloudTrail's <code>GetFederationToken</code> event.
+'`,
+		Platform: stratus.AWS,
+
+		IsIdempotent:               true,
+		MitreAttackTactics:         []mitreattack.Tactic{mitreattack.Persistence},
+		PrerequisitesTerraformCode: tf,
+		Detonate:                   detonate,
+	})
+}
+
+func detonate(params map[string]string, providers stratus.CloudProviders) error {
+	accessKeyID := params["access_key_id"]
+	secretAccessKey := params["secret_access_key"]
+
+	if accessKeyID == "" || secretAccessKey == "" {
```

Comment:
since this is coming from TF, is there any condition where these would be empty? I suspect this check might be unnecessary